### PR TITLE
[Book_Store] realized JWT authentication

### PR DESCRIPTION
**Pull request by fedorovychh**

### CODE REVIEW

File: src/main/java/com/app/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -41,7 +47,15 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 .authenticated()
                 )
                 .httpBasic(Customizer.withDefaults())
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/config/SecurityConfig.java:51]: Please, define session creation policy.

File: src/main/java/com/app/bookstore/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            @NotEmpty
+            @Email
+            String email,
+            @NotEmpty
+            @Size(min = 8, max = 35)
+            String password
+) {
+
+}
+
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/dto/UserLoginRequestDto.java]: Here are a lot of empty lines. Reduce it count.

File: src/main/java/com/app/bookstore/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public String generateToken(String email) {
+        return Jwts.builder()
+                .setSubject(email)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secretKey)
+                .compact();
+    }
+
+    public boolean isValidToken(String token) {
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/security/JwtUtil.java]: Please, rename. For example isTokenValid

File: src/main/java/com/app/bookstore/config/SecurityConfig.java [hunk 4]
```diff
@@ -1,16 +1,20 @@
 package com.app.bookstore.config;
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/config/SecurityConfig.java]: It's better to place this file in a `security` package.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just a few points to consider.

**Comment by fedorovychh** [comment]
Added new changes because of updated structure from previous PR

File: src/main/java/com/app/bookstore/dto/UserLoginRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+public record UserLoginRequestDto(
+            @NotEmpty
+            @Email
+            String email,
+            @NotEmpty
+            @Size(min = 8, max = 35)
+            String password
+) {}
+
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/dto/UserLoginRequestDto.java]:
```suggestion
) {
}
```

**Review by bhdnchui — APPROVED** [review]
LGTM

**Pull request merged**